src/histogram_equalization.jl

```diff
@@ -0,0 +1,217 @@
+    image_min, image_max = minimum(img), maximum(img)
+    normalized_image = (img .- image_min) / (image_max - image_min)
+
+    # Step 2: Apply adaptive histogram equalization. equalize_adapthist handles the tiling to 1/8 of the image size (equivalent to 8x8 blocks in MATLAB)
+    equalized_image = sk_exposure.equalize_adapthist(
+        normalized_image,
+        clip_limit=clip,  # Equivalent to MATLAB's 'ClipLimit'
+        nbins=nbins         # Number of histogram bins. 255 is used to match the default in MATLAB script
+    )
+
+    # Step 3: Rescale the image back to the original range [image_min, image_max]
+    final_image = sk_exposure.rescale_intensity(equalized_image, in_range="image", out_range=(image_min, image_max))
+
+    # Convert back to the original data type if necessary
+    final_image = to_uint8(final_image)
+
+    return final_image
+end
+
+"""
+    getrbc_channels(img)
+
+Get the RBC (Red, Blue, and Green) channels of an image.
+
+# Arguments
+- `img`: The input image.
+
+# Returns
+An m x n x 3 array the Red, Blue, and Green channels of the input image.
+
+"""
+function getrbc_channels(img)
+    # TODO: might be able to use channelview instead
+    redc = red.(img) * 255
+    greenc = green.(img) * 255
+    bluec = blue.(img) * 255
+
+    return cat(redc, greenc, bluec, dims=3)
+end
+
+
+"""
+    conditional_histeq(true_color_img, clouds, landmask, rblocks::Int=8, cblocks::Int=6, entropy_threshold::AbstractFloat=4.0, white_threshold::AbstractFloat=25.5, white_fraction_threshold::AbstractFloat=0.4)
+
+Performs conditional histogram equalization on a true color image.
+
+# Arguments
+- `true_color_img`: The true color image to be equalized.
+- `false_color_image`: The false color image used to determine the regions to equalize.
+- `landmask`: The land mask indicating the land regions in the image.
+- `rblocks`: The number of row-blocks to divide the image into for histogram equalization. Default is 8.
+- `cblocks`: The number of column-blocks to divide the image into for histogram equalization. Default is 6.
+- `entropy_threshold`: The entropy threshold used to determine if a block should be equalized. Default is 4.0.
+- `white_threshold`: The white threshold used to determine if a pixel should be considered white. Default is 25.5.
+- `white_fraction_threshold`: The white fraction threshold used to determine if a block should be equalized. Default is 0.4.
+
+# Returns
+The equalized true color image.
+
+"""
+function conditional_histeq(
+    true_color_img,
+    false_color_image,
+    landmask,
+    rblocks::Int=8,
+    cblocks::Int=6,
+    entropy_threshold::AbstractFloat=4.0,
+    white_threshold::AbstractFloat=25.5,
+    white_fraction_threshold::AbstractFloat=0.4,
+    prelim_threshold=110.0,
+    band_7_threshold=200.0,
+    band_2_threshold=190.0,)
+
+    # Get image for basis of conditional adaptive histogram equalization
+    clouds_red = get_red_channel_cloud_cae(
```

Comment:
Does this need to call the private function `_get_red_channel_cloud_cae`? That could be the source of the CI error?

Reply:
Indeed!